src/Disks/ObjectStorages/Web/MetadataStorageFromStaticFilesWebServer.cpp

```diff
@@ -53,7 +54,23 @@ const std::string & MetadataStorageFromStaticFilesWebServer::getPath() const
 
 bool MetadataStorageFromStaticFilesWebServer::exists(const std::string & path) const
 {
-    return object_storage.files.contains(path);
+    fs::path fs_path(path);
+    if (fs_path.has_extension())
+        fs_path = fs_path.parent_path();
+
+    initializeIfNeeded(fs_path, false);
+
+    if (object_storage.files.empty())
+        return false;
+
+    if (object_storage.files.contains(path))
+        return true;
+
+    for (const auto & [object_path, _] : object_storage.files)
+        if (startsWith(object_path, path))
```

Comment:
I don't understand, why "starts_with"?

Reply:
`object_storage.files` contains files + directories only inside `metadata_path / uuid_3_digit / uuid /`, e.g with previous code it would say that `metadata_path` does not exist if we called `exists(<metadata_path>)`. But with current code this is works ok without this change, but there is another PR, which is not related to diskWeb, but to it's read-only nature, and therefore affects this code a little. And there it becomes important to have a more correct behaviour.